### PR TITLE
Lims 3710

### PR DESCRIPTION
**Pull request by mazenmmbadr**

### CODE REVIEW

File: ui_testing/pages/testplan_page.py [hunk 1]
```diff
@@ -44,17 +45,19 @@ def set_test_plan(self, name='', random=False):
...
+    def set_test_unit(self, test_unit='', add_limits=True, **kwargs):
         self.base_selenium.click('test_plan:next')
         self.base_selenium.click('test_plan:add_test_units')
-        self.base_selenium.select_item_from_drop_down(element='test_plan:test_units', item_text=test_unit)
+        testunit_added_result = self.base_selenium.select_item_from_drop_down(element='test_plan:test_units', item_text=test_unit)
+        if testunit_added_result == False:
+            return False
         self.base_selenium.click('test_plan:add')
-        if 'upper' in kwargs:
+        if add_limits and 'upper' in kwargs:
```
Comment (0xIslamTaha) [inline ui_testing/pages/testplan_page.py]: no need to do that as the old code only execute if the 'upper' in kwatgs

File: ui_testing/pages/testplan_page.py [hunk 2]
```diff
@@ -44,17 +45,19 @@ def set_test_plan(self, name='', random=False):
...
             self.base_selenium.LOGGER.info(' set upper : {}'.format(kwargs['upper']))
             elems = self.base_selenium.find_elements('general:col_6')
             upper = self.base_selenium.find_element_in_element(source=elems[4], destination_element='general:input')
             upper.send_keys(kwargs['upper'])
-        if 'lower' in kwargs:
+        if add_limits and  'lower' in kwargs:
```
Comment (0xIslamTaha) [inline ui_testing/pages/testplan_page.py]: no need to do that as the old code only execute if the 'lower' in kwatgs

File: ui_testing/pages/testplan_page.py [hunk 3]
```diff
@@ -67,7 +70,7 @@ def get_test_unit_limits(self):
         lower = self.base_selenium.find_element_in_element(source=elems[5], destination_element='general:input')
         return upper.get_attribute('value'), lower.get_attribute('value')
 
-    def create_new_test_plan(self, name='', material_type='', article='', test_unit='', **kwargs):
+    def create_new_test_plan(self, name='', material_type='', article='', test_unit='', add_limits=True, **kwargs):
```
Comment (0xIslamTaha) [inline ui_testing/pages/testplan_page.py]: no need

File: ui_testing/pages/testplan_page.py [hunk 4]
```diff
@@ -90,10 +93,9 @@ def create_new_test_plan(self, name='', material_type='', article='', test_unit=
 
         if test_unit:
             self.base_selenium.LOGGER.info('With {} test unit'.format(test_unit))
-            self.set_test_unit(test_unit=test_unit, **kwargs)
-            self.save(save_btn='test_plan:save')
-        else:
-            self.save()
+            self.set_test_unit(test_unit=test_unit,add_limits=add_limits , **kwargs)
```
Comment (0xIslamTaha) [inline ui_testing/pages/testplan_page.py]: no need

File: ui_testing/testcases/basic_tests/test04_testunits.py [hunk 5]
```diff
@@ -1144,3 +1144,32 @@ def test031_editing_limit_of_quantification_fields_should_affect_table_and_versi
...
+        LIMS-3710
+
+        analysis check is postponed until analysis page is created.
+        """
+
+        self.base_selenium.LOGGER.info('archive random testunits')
+        selected_test_units_data = self.test_unit_page.get_random_test_units_row()
+        row_data = self.base_selenium.get_row_cells_dict_related_to_header(row=selected_test_units_data)
+
+        self.test_unit_page.archive_selected_test_units()
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test04_testunits.py]: when did u select it?

Comment (mazenmmbadr) [reply ui_testing/testcases/basic_tests/test04_testunits.py]: this was unpushed commit

File: ui_testing/pages/testplan_page.py (hunk 1 above)
Comment (mazenmmbadr) [reply ui_testing/pages/testplan_page.py]: done

File: ui_testing/pages/testplan_page.py (hunk 2 above)
Comment (mazenmmbadr) [reply ui_testing/pages/testplan_page.py]: done

File: ui_testing/pages/testplan_page.py (hunk 4 above)
Comment (mazenmmbadr) [reply ui_testing/pages/testplan_page.py]: done

File: ui_testing/pages/testplan_page.py (hunk 3 above)
Comment (mazenmmbadr) [reply ui_testing/pages/testplan_page.py]: done

**Pull request merged**